schemes/musica/tuvx/musica_ccpp_tuvx.F90

```diff
@@ -134,11 +143,7 @@ subroutine tuvx_init(vertical_layer_dimension, vertical_interface_dimension, &
           wavelength_grid, temperature_profile, surface_albedo_profile )
 
     grids => tuvx%get_grids( error )
-    if (has_error_occurred( error, errmsg, errcode )) then
-      deallocate( tuvx )
-      tuvx => null()
-      return
-    end if
+    if (has_error_occurred( error, errmsg, errcode )) return
```

Comment:
I removed any deallocations that are already handled by the `tuvx_final()` function (assuming this gets called by the host model if there is an error). But, I'm not opposed to putting explicit deallocations in the error handling blocks, it just makes for more code. What do you prefer?

Reply:
I think the code will be cleaner if we could do that. From the previous discussion with @nusbaume, I learned that in CAM-SIMA, the model run is aborted as soon as a error code other than zero is raised. So at least for now, it will never reaches the final phase. But he also mentioned that if we want different behavior for failures in MUSICA, they could implement alternative fallbacks in CAM-SIMA, such as jumping straight to the final call. Would you prefer that approach? We could bring it up in the next CAM-SIMA meeting.

Reply:
I have no strong preference. What if I just called the `tuvx_final()` function inside these error blocks, just to be sure everything gets deallocated?

Reply:
One issue with trying to clean up before the job ends is that often, an error occurs on only one or a subset of atmosphere tasks. Thus, the job is killed without many or most tasks being aware of the situation. Current architectures do not give us a lot of automatic alternatives that do not sacrifice performance. I think cleanup is great in principle and perhaps there is a way to invoke it on all tasks so I think just having it all in the final routine is most likely to be useful in a CAM-SIMA environment.

Reply:
I think in general calling the finalize phase before CAM-SIMA aborts is a good idea and should be very easy to do on the CAM-SIMA side, so I went ahead and added an issue for it, which you can find here:

https://github.com/ESCOMP/CAM-SIMA/issues/322

So for this PR if you want to move all of the deallocation steps and any other cleanup that you want to do into `tuvx_final` then CAM-SIMA should automatically run that before the program stops (at least once the relevant CAM-SIMA PR is made).  Hope that helps!